Components/FPSCounter.cs

```diff
@@ -0,0 +1,82 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public static class FPSCounter
+{
+    public static int s_smoothingWindowSize = 100;
+
+    private static Queue<float> s_measures = new Queue<float>(s_smoothingWindowSize);
+    private static int s_lastProcessedFrame = -1;
+    private static int s_lastCalculatedFrame = -1;
+
+    private static float s_minFrameTime;
+    private static float s_maxFrameTime;
+    private static float s_averageFrameTime;
+
+    public static float MinFrameTime
+    {
+        get
+        {
+            _Process();
+            return s_minFrameTime;
+        }
+    }
+
+    public static float MaxFrameTime
+    {
+        get
+        {
+            _Process();
+            return s_maxFrameTime;
+        }
+    }
+
+    public static float AverageFrameTime
+    {
+        get
+        {
+            _Process();
+            return s_averageFrameTime;
+        }
+    }
+
+    private static void _Process()
```

Comment:
This doesn't call `UpdateManually()` - why? Data will be delayed if we're asking for `AverageFrameTime` every frame, for example, and our `UpdateManually()` call happens after getting to that field. So to guarantee that we're always getting fresh data we'd have to either fiddle with Script Execution Order, or preface **every** point of access to `.xxxFrameTime` property with `UpdateManually()` - it's just lost elegance